library/date/deconstruct_keys_spec.rb

```diff
@@ -0,0 +1,31 @@
+require_relative '../../spec_helper'
+require 'date'
+
+ruby_version_is "3.2" do
+  describe "Date#deconstruct_keys" do
+    it "returns whole hash for nil as an argument" do
+      d = Date.new(2022, 10, 5)
+      d.deconstruct_keys(nil).should == { year: 2022, month: 10, day: 5, yday: 278, wday: 3 }
+    end
+
+    it "returns only specified keys" do
+      d = Date.new(2022, 10, 5)
+      d.deconstruct_keys([:year, :month]).should == { year: 2022, month: 10 }
+    end
+
+    it "requires one argument" do
+      -> {
+        Date.new(2022, 10, 5).deconstruct_keys
+      }.should raise_error(ArgumentError)
+    end
+
+    it "accepts only array or nil as an argument" do
+      d = Date.new(2022, 10, 5)
+
+      -> { d.deconstruct_keys(1) }.should raise_error(TypeError)
+      -> { d.deconstruct_keys("asd") }.should raise_error(TypeError)
+      -> { d.deconstruct_keys(:x) }.should raise_error(TypeError)
+      -> { d.deconstruct_keys({}) }.should raise_error(TypeError)
+    end
```

Comment:
**suggestion:** It's useful to have exception messages specified in tests. Unless they are cryptic or change often in different Ruby versions.